app/controllers/categories_controller.rb

```diff
@@ -128,33 +128,37 @@ def quick_create
   end
 
   private
-    # Use callbacks to share common setup or constraints between actions.
-    def set_category
-      begin
-        @category = Category.find(params[:id])
-      rescue
-        if @category.blank?
-          respond_to do |format|
-            format.html { redirect_to categories_path }
-            format.json { head :no_content }
-          end
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_category
+    begin
+      @category = Category.friendly.find(params[:id])
+    rescue
+      if @category.blank?
+        respond_to do |format|
+          format.html { redirect_to categories_path }
+          format.json { head :no_content }
         end
       end
     end
+  end
 
-    def category_params
-      params.require(:category).permit(:name, :description, :userid)
-    end
-
-    def is_viewer(moment, strategy, category)
-      if !strategy.blank? && Strategy.where(id: strategy).exists? && Strategy.where(id: strategy).first.viewers.include?(current_user.id)
-        return true
-      elsif !moment.blank?
-        if Moment.where(id: moment).exists? && Moment.where(id: moment).first.category.include?(category.id) && Moment.where(id: moment).first.viewers.include?(current_user.id) && are_allies(moment.userid, current_user.id)
-          return true
-        end
-      end
+  def category_params
+    params.require(:category).permit(:name, :description, :userid)
+  end
 
-      return false
+  def is_viewer(moment, strategy, category)
```

Comment:
this seems like a good thing to pull out into its own class. there are a lot of other helper methods that seem to do a similar thing (in application, moments, and mood controllers) so it would be cool to have a Viewer class or something like that. also these variables are actually IDs so it would be great to be more explicit about that (e.g. rename to `moment_id`, etc)